imap_processing/codice/codice_l1a.py

```diff
@@ -741,6 +880,115 @@ def log_dataset_info(datasets: dict[int, xr.Dataset]) -> None:
         )
 
 
+def reshape_de_data(
+    packets: xr.Dataset, decompressed_data: list[list[int]], num_priorities: int
+) -> dict[str, list[np.ndarray]]:
+    """
+    Reshape the decompressed direct event data into CDF-ready arrays.
+
+    Parameters
+    ----------
+    packets : xarray.Dataset
+        Dataset containing the packets, needed to determine priority order
+        and data quality.
+    decompressed_data : list[list[int]]
+        The decompressed data to reshape, in the format <epoch>[<priority>[<event>]].
+    num_priorities : int
+        The number of priorities in the data product (differs between CoDICE-Lo
+        and CoDICE-Hi).
+
+    Returns
+    -------
+    data : dict[str, list[np.ndarray]]
+        The reshaped, CDF-ready arrays. The keys of the dictionary represent the
+        CDF variable names, and the values represent the data.
+    """
+    # Dictionary to hold all the (soon to be restructured) direct event data
+    data: dict[str, list[np.ndarray]] = {}
+
+    # Create blank arrays for each priority and field to store the data
+    # We also need arrays to hold number of events and data quality
+    for i in range(num_priorities):
+        for field in constants.LO_DE_BIT_STRUCTURE:
+            if field not in ["Priority", "Spare"]:
+                data[f"P{i}_{field}"] = []
+        data[f"P{i}_NumEvents"] = []
+        data[f"P{i}_DataQuality"] = []
+
+    # decompressed_data is one large list of values of length
+    # (<number of epochs> * <8 priorities>)
+    # Chunk the data into each epoch/priority combination
```

Comment:
Rather than appending/updating lists here can you make the initial data arrays with 0s of the expected shape? (nepoch, 10000) and then as you're going through the loops just fill in the proper index value?